### PR TITLE
[ORCT0-134] - filtering for new api

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- added possibility to filter fetched data
Example:

Client use url: 
<pre> <code> const URL  = .../?filter[or][field1][gt]=10&filter[or][field1][lt]=3&filter[or][field1][_goperator]=or&filter[or][filed2][like]=LHC_%pass&filter[or][filed2][notLike]=LHC_c%  
</code></pre>

The url will be parsed by qs in express to:  
<pre> <code> const query = {
      "filter": {
          "or": {
              "field1": {
                  "gt": "10",
                  "lt": "3",
                  "_goperator": "or"
              },
              "filed2": {
                  "like": "LHC_%pass",
                  "notLike": "LHC_c%"
              }
      }
} </code></pre>

query.filter is being passed as an argument to filterToSequelizeWhereClause function
which returns sequelize-compatible where-clause 
<pre><code> const whereClause = {
 [Symbol(or)]: {
   field1: {
     [Symbol(or)]: {
       [Symbol(gt)]: '10',
       [Symbol(lt)]: '3'
     }
   },
   filed2: {
     [Symbol(and)]: {
       [Symbol(like)]: 'LHC_%pass',
       [Symbol(notLike)]: 'LHC_c%'
     }
   }
 }
} </code></pre>

What is equivalent to sql: 
<pre><code>  WHERE (field1 > 10 OR field1 < 3) OR (field2 like 'LHC_%pass' AND field2 NOT LIKE 'LHC_c%') </pre></code>

##!!! For entity Run filtering with columns runDuration, endTime, startTime is not supported because they are virtual fields, which with sequelize quering is not possible

Notable changes for developers:
- details of previous in docs of mentioned function `filterToSequelizeWhereClause`
- func `validateDTO` renamed to `validateDtoOrRepondOnFailure`
- object `STDEntiryDTO` renamed to  `stdDataRequestDTO`
- introduced `controllerHandlerWrapper` for handling all uncaught errors during processing clients requests 
- tests

### CODE REVIEW

File: app/lib/server/utilities/filterParser.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            }
+        }),
+);
+
+/**
+ * Transform filter object from http request.query to sequelize where object
+ * Assuming that the filter object can be considered as tree, then each root -> leaf,
+ * 1. path can be described using one of two patterns
+ * (dots separate nodes, words written uppercase are non-terminal, (as in formal gramatics terminology)
+ * and that written lowercase are terminal, e.g. operator not is termianl):
```
Comment (Ehevi) [inline app/lib/server/utilities/filterParser.js:61]: Typo (`termianl` -> `teminal`)

Comment (xsalonx) [reply app/lib/server/utilities/filterParser.js:61]: ok

File: app/lib/server/utilities/filterParser.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        filter = this.insertSentinel(filter);
+        filter = this.transformHelper(filter);
+        filter = this.removeSentinel(filter);
+        return filter;
+    }
+}
+
+/**
+ * Transform filter object from http request.query to sequelize where-clause object
+ * Considering that the filter object as a tree, then each root -> leaf path can be described as ,
```
Comment (Ehevi) [inline app/lib/server/utilities/filterParser.js:93]: `Considering the filter object is a tree, each root -> leaf path can be described as`

Comment (xsalonx) [reply app/lib/server/utilities/filterParser.js:93]: ok

**Pull request merged**